### DISCUSSION TITLE
Ensure the 'inputs' field in /chat-messages takes effect every time

### DISCUSSION THREAD

**Discussion by crazywoola** [langgenius/dify#10652] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

Regarding this PR request, we are looking for the feedback from the community. Either testing it or feedbacks are welcome.

https://github.com/langgenius/dify/pull/7955

If you support this feature, please leave a 👍 in this discussion. Or leave 👀 if you don't support it or you have something to say.

**Comment by Wu-Jiayang** [comment]
I would like to ask why you decided to fix the 'inputs' field during the design?

**Reply by crazywoola** [reply]
Maybe like a snapshot, I think in a very early stage, we were putting more effort in completion app, so it's more like a one time workflow at this moment.

**Reply by Wu-Jiayang** [reply]
I see. This is my first time submitting a PR to an open-source project. If you decide to merge it, will I become a Contributor?

**Comment by liuzixiao666** [comment]
At present, we need different responses for visits from different customers at different times and locations, so our agent needs to obtain this part of the variables in advance. Previously, our variables were in the input variable, but we found that we can only guarantee the variables at the beginning of the session. If the state changes later, we still cannot make the same response in a timely manner. If we follow the feedback in real time, we must open a new session, which will cause short-term memory failure. For example, when a user asks about a certain product, we introduce the price and other information of the product. During this process, our product may have been sold out, but the agent cannot know. If the user still wants to ask, the agent will inform them that they can purchase it. It is not realistic to store the real-time status of products in a knowledge base, and embedding it in a workflow using HTTP requests requires ensuring the correctness of many parameters, which places high demands on our development

**Reply by liuzixiao666** [reply]
> At present, we need different responses for visits from different customers at different times and locations, so our agent needs to obtain this part of the variables in advance. Previously, our variables were in the input variable, but we found that we can only guarantee the variables at the beginning of the session. If the state changes later, we still cannot make the same response in a timely manner. If we follow the feedback in real time, we must open a new session, which will cause short-term memory failure. For example, when a user asks about a certain product, we introduce the price and other information of the product. During this process, our product may have been sold out, but the agent cannot know. If the user still wants to ask, the agent will inform them that they can purchase it. It is not realistic to store the real-time status of products in a knowledge base, and embedding it in a workflow using HTTP requests requires ensuring the correctness of many parameters, which places high demands on our development

At the same time, I accept that some fields only need to be entered once, such as customer basic information, and some fields support real-time changes, such as the current customer status

**Comment by dreamTrainer** [comment]
I need this feature.

**Comment by liuxin19971202** [comment]
Is this feature finally going to be released? I am still using FastGPT and hope to update dify as soon as possible!!!

**Comment by HaiShin** [comment]
I hope that each time the /chat-message interface is called, new inputs can be used instead of a one-time input. Because I hope that it can obtain the page cookies in real time to determine whether the login has expired. Or it can provide more intelligent Q&A by judging the content of the page.

**Comment by mihit** [comment]
https://github.com/langgenius/dify/issues/11140
I asked a question here as well.
Frankly speaking, I mistook the current implementation for a bug. Because the interface looks different from the behavior we expect.
I think that inputs that you want to persist should be isolated in a field for that purpose, like "persist_inputs", and any request that tries to overwrite them should raise an Exception.

**Reply by Wu-Jiayang** [reply]
My thought is exactly the same as you!

**Reply by Wu-Jiayang** [reply]
You can see my PR https://github.com/langgenius/dify/pull/7955
I tested it on v0.11.1. It works well.

**Comment by liuzixiao666** [comment]
> ### Self Checks
> * [x]  I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
> * [x]  I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
> * [x]  [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
> * [x]  Please do not modify this template :) and fill in all the required fields.
> 
> ### Content
> Regarding this PR request, we are looking for the feedback from the community. Either testing it or feedbacks are welcome.
> 
> #7955
> 
> If you support this feature, please leave a 👍 in this discussion. Or leave 👀 if you don't support it or you have something to say.

Brother, has there been a result of the discussion? I really need this feature. From 0.11.0 to 0.13.0, it seems that none of them have this feature.

**Comment by cubxxw** [comment]
My dify does not decide to update, I have been waiting for this PR merge until now
...

**Reply by liuzixiao666** [reply]
me too

**Comment by zdq5337** [comment]
hope to merge

**Comment by liuzixiao666** [comment]
Just wait, wait, wait again

**Comment by Wu-Jiayang** [comment]
Hello? Is anyone here?

**Comment by Wu-Jiayang** [comment]
My friend saw you in CES
![07640f705eee6b1486bf2edbb70cf25](https://github.com/user-attachments/assets/bb9f6c27-ff4b-43a5-803c-7ba7d0fea2bf)
